client/simplestreams_images.go

```diff
@@ -84,6 +84,11 @@ func (r *ProtocolSimpleStreams) GetImageFile(fingerprint string, req ImageFileRe
 
 		size, err := shared.DownloadFileSha256(r.http, r.httpUserAgent, req.ProgressHandler, req.Canceler, filename, url, sha256, target)
 		if err != nil {
+			// Handle cancelation
+			if err.Error() == "net/http: request canceled" {
```

Comment:
Did you already check if there's something better than comparing the err msg to "net/http: request canceled" ?

Also, the http.Request.Cancel channel that is used by the underlying code is deprecated. One should use context.Context these days (and that does provide a way to check if an error is due to cancellation).